src/sentry/monitors/endpoints/organization_monitor_index_stats.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+from collections import OrderedDict, defaultdict
+from datetime import datetime, timedelta
+from typing import List, MutableMapping
+
+from django.db.models import Count, DateTimeField, Func
+from django.db.models.functions import Extract
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import StatsMixin, region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.helpers.environments import get_environments
+from sentry.monitors.models import CheckInStatus, Monitor, MonitorCheckIn
+from sentry.utils.dates import to_timestamp
+
+
+def normalize_to_epoch(timestamp: datetime, seconds: int):
+    """
+    Given a ``timestamp`` (datetime object) normalize to an epoch timestamp.
+
+    i.e. if the rollup is minutes, the resulting timestamp would have
+    the seconds and microseconds rounded down.
+    """
+    epoch = int(to_timestamp(timestamp))
+    return epoch - (epoch % seconds)
+
+
+@region_silo_endpoint
+class OrganizationMonitorIndexStatsEndpoint(OrganizationEndpoint, StatsMixin):
+    # TODO(epurkhiser): probably convert to snuba
+    def get(self, request: Request, organization) -> Response:
+        args = self._parse_args(request)
+
+        start = normalize_to_epoch(args["start"], args["rollup"])
+        end = normalize_to_epoch(args["end"], args["rollup"])
+
+        monitor_slugs: List[str] = request.GET.getlist("monitor")
+
+        tracked_statuses = [
+            CheckInStatus.OK,
+            CheckInStatus.ERROR,
+            CheckInStatus.MISSED,
+            CheckInStatus.TIMEOUT,
```

Comment:
we could perhaps think about include IN_PROGRESS as well to show if a job is currently running, perhaps after we play around with it more